### ISSUE TITLE
feat: support mute

### ISSUE THREAD

**Issue by Ushie** [ReVanced/revanced-helper#19]
Superseding #15 as the discussion there has gotten too long and hard to follow

Support Mute is now implemented in ReVanced Helper, under the name of "exile", but it's exclusively a slash command, given the context of the motivation for this command to exist in the first place, it's more ideal for it to be available in the context menu, that'd be easier for rapid fire and it also skips having to manually specify a message to delete

I'm also not sure whether this was implemented or not, but the message that was specified should also be moved to the support channel with this format:

"@Ushie said: why is revanced stopping at 1:09"

This ensures that the message is kept and the user doesn't have to write their message twice, and also helps in telling the user where they should go as they now have a ping in the support channel

The commands would follow this structure: 
"Support mute" in context menu, which does the following:
1. give the user the support muted role
2. moves the user's message to #support in the structure shown above
3. delete the initial message in chat

?smute as a command that depends on replying to messages:
1. the same 3 steps above
2. delete the message containing the ?smute command

Why reply? it's faster and specifies the message that should be moved and deleted

**Comment by SodaWithoutSparkles** [comment]
If rapid fire is the goal, then slash commands would be quicker as you can copy and paste, then ping the user. The distance for your finger to travel is shorter.

But we cant select which msg to redirect.

**Comment by oSumAtrIX** [comment]
> s more ideal for it to be available in the context menu, that'd be easier for rapid fire and it also skips having to manually specify a message to delete

On mobile it may be faster to use slash commands

**Comment by taku-nm** [comment]
>Why reply?

It would sure be nice to implement a regular text command with reply parsing.
This was mentioned in the last issue but Reis had [this](https://github.com/ReVanced/revanced-helper/issues/15#issuecomment-1667709203) to say

**Comment by Ushie** [comment]
> > Why reply?
> 
> It would sure be nice to implement a regular text command with reply parsing. This was mentioned in the last issue but Reis had [this](https://github.com/ReVanced/revanced-helper/issues/15#issuecomment-1667709203) to say

That is because it was being referred to as a prefix command, with parameters and everything, in this case, that's not necessary as replying is necessary, the bot can get the necessary details from the message that was replied to

**Comment by taku-nm** [comment]
Proposal
3 Options to execute this command:
(exile or smute doesn't matter)

(replied message) ?smute [no param]
/smute [user] [msg count]
context menu: support-mute [no param]

**Comment by SodaWithoutSparkles** [comment]
I propose using stopit instead of smute to make it more interesting.

_stop it, get some help_

**Comment by oSumAtrIX** [comment]
When does the bot remove the role again?

**Comment by taku-nm** [comment]
I'd suggest 5 or 10 minutes

**Comment by oSumAtrIX** [comment]
Should be configurable via the bot configs.

**Comment by taku-nm** [comment]
Just to keep track of the current implementation and what's missing from it:

- Bot should delete the original message it copied to support (cut and paste principle)
- Bots should no longer reply in cafe with help
- Bot should not be able to exile itself
- The logging of exile and unexile actions should happen in the logs thread

**Comment by oSumAtrIX** [comment]
@taku-nm I've added checkboxes

**Comment by oSumAtrIX** [comment]
@taku-nm Why should bots not reply in cafe with help? Or what are you referring to?

**Comment by taku-nm** [comment]
> @taku-nm Why should bots not reply with help? Or what are you referring to?

It should not reply in cafe because why should it? User asks a question, gets exiled and can't read the bot reply anyways.
One might make the bot interpret the reply it's cuting and pasting and automatically reply to it in support chat.
@ User
Your question: 
Proposed answer:

**Comment by SodaWithoutSparkles** [comment]
I think he is refering to the "you do not have permission to do this" thing.

I propose, instead of sending a message, just react a 🙄 or ❌.

**Comment by reisxd** [comment]
The bot does log in logs and the bot does not reply in any channel unless you run a slash command. Also, I don't think the bot replying "You don't have the permission to do that" is much of a problem to add unnecessary code. It does not cover uo your whole screen.

**Comment by oSumAtrIX** [comment]
> It should not reply in cafe because why should it?

To retain history/context. Messages randomly disappearing will be confusing, instead it should show a message "Moved to X" or similar.

**Comment by oSumAtrIX** [comment]
> What? I think we're talking about different things here. I'm saying that the rust bot should no longer reply to questions in cafe as they should not be answered / even asked there in the first place.

The rust bot is not part of this issue and is offtopic. If you expect any changes for the rust bot, please do so in the correct repository.

**Comment by taku-nm** [comment]
I've just checked, this also applies to the helper bot tho. It still replies with with troubleshooting help outside the support channel.

**Comment by oSumAtrIX** [comment]
Are you refering to autoresponses?

**Comment by taku-nm** [comment]
> Are you refering to autoresponses?

Yes

**Comment by oSumAtrIX** [comment]
Specific auto-responses should still be sent in channels like #cafe. However, we are talking about a server-specific configuration, which is out of the scope of this issue.

**Comment by PalmDevs** [comment]
Any updates for this? Can someone summarize the whole discussion to a list of things I need to implement?

**Comment by Ushie** [comment]
https://github.com/ReVanced/revanced-helper/issues/19#issuecomment-1676429790

**Comment by PalmDevs** [comment]
Closed as it is no longer added. If needed again, can be added via the "role presets" feature.
